### PR TITLE
Update logic for retrieving repo language

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Update Logic for retrieving repo language in Verify Changelog Script

### CODE REVIEW

File: eng/common/scripts/Verify-ChangeLog.ps1 [hunk 1]
```diff
@@ -25,7 +25,13 @@ else
   Import-Module (Join-Path $PSScriptRoot modules Package-Properties.psm1)
   if ([System.String]::IsNullOrEmpty($Language))
   {
-    $Language = $RepoName.Substring($RepoName.LastIndexOf('-') + 1)
+    switch -Regex ($RepoName)
```
Comment (weshaggard) [inline eng/common/scripts/Verify-ChangeLog.ps1:28]: If we are going to use a regex I think this should work. 

```
if ($RepoName -match "azure-sdk-for-(?<lang>[^-]*)") 
{
  $Language = $matches["lang"]
}
```

Comment (weshaggard) [reply eng/common/scripts/Verify-ChangeLog.ps1:28]: By the way one of the improvements I've been thinking about is to setup a common ps1 file that we can have in each of the language repo's (outside of the eng\common directory) that can define things like `$global::Language = "<lang>"` That will allow us to parameterize variables and provide language specific methods directly in the language repo but use them in our common scripts so we don't always have to parse or pass them to all our functions.

Comment (chidozieononiwu) [reply eng/common/scripts/Verify-ChangeLog.ps1]: I created an issue to track that https://github.com/Azure/azure-sdk/issues/1544

File: eng/common/scripts/Verify-ChangeLog.ps1 [hunk 2]
```diff
@@ -1,38 +1,42 @@
...
 {
   Import-Module (Join-Path $PSScriptRoot modules Package-Properties.psm1)
   if ([System.String]::IsNullOrEmpty($Language))
   {
-    $Language = $RepoName.Substring($RepoName.LastIndexOf('-') + 1)
+    if ($RepoName -match "azure-sdk-for-(?<lang>[^-]+)")
+    {
+      $Language = $matches["lang"]
+    }
   }
```
Comment (weshaggard) [inline eng/common/scripts/Verify-ChangeLog.ps1:32]: If language isn't set then the below steps will fail in some unknown way so I suggest adding a check here with a clear error message if language isn't set.

File: eng/common/scripts/Verify-ChangeLog.ps1 [hunk 3]
```diff
@@ -1,38 +1,46 @@
...
   {
-    $Language = $RepoName.Substring($RepoName.LastIndexOf('-') + 1)
+    if ($RepoName -match "azure-sdk-for-(?<lang>[^-]+)")
+    {
+      $Language = $matches["lang"]
+    }
+    else
+    {
+      Write-Error "Failed to set Language automatically. Please pass the appropriate Language as a parameter."
+    }
```
Comment (weshaggard) [inline eng/common/scripts/Verify-ChangeLog.ps1:35]: I would also add the exit 1 here.

**Pull request merged**